src/natcap/invest/utils.py

```diff
@@ -732,23 +639,56 @@ def read_csv_to_dataframe(
 
     """
     try:
+        # set index_col=False to force pandas not to index by any column
+        # this is useful in case of trailing separators
+        # we'll explicitly set the index column later on
         dataframe = pandas.read_csv(
-            path, sep=sep, engine=engine, encoding=encoding, **kwargs)
+            path, index_col=False, sep=sep, engine=engine, encoding=encoding, **kwargs)
     except UnicodeDecodeError as error:
         LOGGER.error(
             f'{path} must be encoded as UTF-8 or ASCII')
         raise error
 
+    # strip whitespace from column names
     # this won't work on integer types, which happens if you set header=None
     # however, there's little reason to use this function if there's no header
     dataframe.columns = dataframe.columns.str.strip()
-    if to_lower:
+
+    # convert column names to lowercase
+    if cols_to_lower:
         dataframe.columns = dataframe.columns.str.lower()
+        # if 'to_lower`, case handling is done before trying to access the data.
+        # the columns are stripped of leading/trailing whitespace in
+        # ``read_csv_to_dataframe``, and also lowercased if ``to_lower`` so we only
+        # need to convert the rest of the table.
+        if index_col and isinstance(index_col, str):
+            index_col = index_col.lower()
+        # lowercase column names
+        if usecols:
+            usecols = [col.lower() for col in usecols]
 
-    # Remove values with leading ('^ +') and trailing (' +$') whitespace.
-    # Regular expressions using 'replace' only substitute on strings.
-    dataframe = dataframe.replace(r"^ +| +$", r"", regex=True)
+    # Subset dataframe by columns if desired
+    if usecols:
+        dataframe = dataframe[usecols]
 
+    # Set 'index_col' as the index of the dataframe
+    if index_col:
+        try:
+            dataframe = dataframe.set_index(
+                index_col, drop=False, verify_integrity=True)
+        except KeyError:
+            # If 'index_col' is not a column then KeyError is raised for using
+            # it as the index column
+            LOGGER.error(f"The column '{index_col}' could not be found "
+                         f"in the table {path}")
+            raise
+
+    # convert table values to lowercase
+    if vals_to_lower:
+        dataframe = dataframe.applymap(
+            lambda x: x.lower() if isinstance(x, str) else x)
+
+    # expand paths
```

Comment:
I think there definitely are cases where a table contains a path column that we _don't_ want to lowercase, but other columns that we _do_ want to lowercase. The `read_csv_to_dataframe` approach has not given us the option to distinguish between columns, so we've had to set `to_lower=False` on any table that contains paths. 

When #1328 is implemented, that will give us the chance to apply different processing to each column. For now I don't think this is an issue, because `vals_to_lower` should always be `False` if `expand_paths` are defined.